File: Easy/387 - FirstUniqueCharacterInAString/solution.ts
// Question 387. First Unique Character in a String (https://leetcode-cn.com/problems/first-unique-character-in-a-string/)

/**
 * 执行用时：104 ms, 在所有 TypeScript 提交中击败了 95.52% 的用户
 * 内存消耗：42.2 MB, 在所有 TypeScript 提交中击败了 50.75% 的用户
 *
 * @param {string} s
 * @returns {number}
 */
function firstUniqChar(s: string): number {
  const N: number = s.length;

  for (let i = 0; i < N; i++) {
    const char: string = s[i];
    if (s.indexOf(char) === s.lastIndexOf(char)) {
      return i;
    }
  }

  return -1;
}

File: Easy/387 - FirstUniqueCharacterInAString/Solution.cs
// Question 387. First Unique Character in a String (https://leetcode-cn.com/problems/first-unique-character-in-a-string/)

/// <summary>
/// 执行用时：116 ms, 在所有 C# 提交中击败了 71.09% 的用户
/// 内存消耗：31.6 MB, 在所有 C# 提交中击败了 69.07% 的用户
/// </summary>
public class Solution 
{
    public int FirstUniqChar(string s) 
    {
        int N = s.Length;

        for (int i = 0; i < N; i++)
        {
            char currentChar = s[i];

            if (s.IndexOf(currentChar) == s.LastIndexOf(currentChar))
            {
                return i;
            }
        }

        return -1;
    }
}